tests/test_conda_plugins.py

```diff
@@ -62,13 +79,43 @@ def test_get_token_via_conda_token(handler):
 
 
 @pytest.mark.usefixtures("mocked_token_info")
-def test_get_token_via_keyring(handler):
+def test_get_repo_token_via_keyring(handler):
     token = handler._load_token(
         "https://repo.anaconda.cloud/repo/my-org/my-channel/noarch/repodata.json"
     )
     assert token == "my-test-token-in-token-info"
 
 
+@pytest.mark.usefixtures("mocked_token_info_with_api_key")
+def test_get_api_token_via_keyring(handler, monkeypatch):
+    monkeypatch.setenv("ANACONDA_AUTH_USE_UNIFIED_API_KEY", "True")
+    token = handler._load_token(
+        "https://repo.anaconda.cloud/repo/my-org/my-channel/noarch/repodata.json"
+    )
+    assert token == "my-test-api-key"
+
+@pytest.mark.usefixtures("mocked_token_info")
+def test_auth_handler_call_sets_authorization_header_repo_token(handler, monkeypatch):
+    request = PreparedRequest()
+    request.url = "https://repo.anaconda.cloud/repo/my-org/my-channel/noarch/repodata.json"
+    request.headers = {}
+
+    modified_request = handler(request)
+
+    assert modified_request.headers["Authorization"] == "token my-test-token-in-token-info"
+
+@pytest.mark.usefixtures("mocked_token_info_with_api_key")
+def test_auth_handler_call_sets_authorization_header_api_key(handler, monkeypatch):
+    monkeypatch.setenv("ANACONDA_AUTH_USE_UNIFIED_API_KEY", "True")
+
+    request = PreparedRequest()
+    request.url = "https://repo.anaconda.cloud/repo/my-org/my-channel/noarch/repodata.json"
+    request.headers = {}
+
+    modified_request = handler(request)
+
+    assert modified_request.headers["Authorization"] == "token my-test-api-key"
```

Comment:
We discovered recently that not all clients will support the "token" prefix for repo access (e.g. Artifactory mirrors), so we should use `Bearer` here instead. This will require a bit of manipulation of the header-building code.

If that is excessive within the scope of this PR, we can isolate that to a different PR.